op-chain-ops/deployer/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,206 @@
+	}
+	defer func() {
+		if err := cleanup(); err != nil {
+			lgr.Warn("failed to clean up artifacts", "err", err)
+		}
+	}()
+
+	l1Client, err := ethclient.Dial(cfg.L1RPCUrl)
+	if err != nil {
+		return fmt.Errorf("failed to connect to L1 RPC: %w", err)
+	}
+
+	chainID, err := l1Client.ChainID(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get chain ID: %w", err)
+	}
+	chainIDU64 := chainID.Uint64()
+
+	superCfg, err := opcm.SuperchainFor(chainIDU64)
+	if err != nil {
+		return fmt.Errorf("error getting superchain config: %w", err)
+	}
+	standardVersionsTOML, err := opcm.StandardVersionsFor(chainIDU64)
+	if err != nil {
+		return fmt.Errorf("error getting standard versions TOML: %w", err)
+	}
+	opcmProxyOwnerAddr, err := opcm.ManagerOwnerAddrFor(chainIDU64)
+	if err != nil {
+		return fmt.Errorf("error getting superchain proxy admin: %w", err)
+	}
+
+	signer := opcrypto.SignerFnFromBind(opcrypto.PrivateKeySignerFn(cfg.privateKeyECDSA, chainID))
+	chainDeployer := crypto.PubkeyToAddress(cfg.privateKeyECDSA.PublicKey)
+
+	lgr.Info("deploying OPCM", "release", cfg.ContractsRelease)
+
+	var dio opcm.DeployImplementationsOutput
+	err = pipeline.CallScriptBroadcast(
+		ctx,
+		pipeline.CallScriptBroadcastOpts{
+			L1ChainID:   chainID,
+			Logger:      lgr,
+			ArtifactsFS: artifactsFS,
+			Deployer:    chainDeployer,
+			Signer:      signer,
+			Client:      l1Client,
+			Broadcaster: pipeline.KeyedBroadcaster,
+			Handler: func(host *script.Host) error {
+				// We need to etch the Superchain addresses so that they have nonzero code
+				// and the checks in the OPCM constructor pass.
+				superchainConfigAddr := common.Address(*superCfg.Config.SuperchainConfigAddr)
+				protocolVersionsAddr := common.Address(*superCfg.Config.ProtocolVersionsAddr)
+				addresses := []common.Address{
+					superchainConfigAddr,
+					protocolVersionsAddr,
+				}
+				for _, addr := range addresses {
+					host.ImportAccount(addr, types.Account{
+						Code: []byte{0x00},
+					})
+				}
+
+				var salt common.Hash
+				_, err = rand.Read(salt[:])
+				if err != nil {
+					return fmt.Errorf("failed to generate CREATE2 salt: %w", err)
+				}
+
+				dio, err = opcm.DeployImplementations(
+					host,
+					opcm.DeployImplementationsInput{
+						Salt:                            salt,
+						WithdrawalDelaySeconds:          big.NewInt(604800),
+						MinProposalSizeBytes:            big.NewInt(126000),
+						ChallengePeriodSeconds:          big.NewInt(86400),
+						ProofMaturityDelaySeconds:       big.NewInt(604800),
+						DisputeGameFinalityDelaySeconds: big.NewInt(302400),
+						Release:                         cfg.ContractsRelease,
+						SuperchainConfigProxy:           superchainConfigAddr,
+						ProtocolVersionsProxy:           protocolVersionsAddr,
+						OpcmProxyOwner:                  opcmProxyOwnerAddr,
+						StandardVersionsToml:            standardVersionsTOML,
+						UseInterop:                      false,
```

Comment:
Is the plan to eventually make these configurable for different networks? If so, a todo comment would be helpful.

Reply:
Or would it make sense to reuse the values set in deploy_config.go?

Reply:
Yes, I'll eventually make this configurable. We'll set sane defaults in deploy_config.go, then let users override them.